gtsfm/frontend/detector_descriptor/sift_colmap.py

```diff
@@ -0,0 +1,46 @@
+"""SIFT Detector-Descriptor from Colmap.
+
+Ref: 
+
+Authors: Ayush Baid
+"""
+from typing import Tuple
+
+import numpy as np
+import pycolmap
+
+import gtsfm.utils.images as image_utils
+from gtsfm.common.image import Image
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.detector_descriptor.detector_descriptor_base import DetectorDescriptorBase
+
+
+class SIFTColmap(DetectorDescriptorBase):
+    """SIFT detector-descriptor using OpenCV's implementation."""
+
+    def detect_and_describe(self, image: Image) -> Tuple[Keypoints, np.ndarray]:
+        """Perform feature detection as well as their description.
+
+        Refer to detect() in DetectorBase and describe() in DescriptorBase for details about the output format.
+
+        Args:
+            image: the input image.
+
+        Returns:
+            Detected keypoints, with length N <= max_keypoints.
+            Corr. descriptors, of shape (N, D) where D is the dimension of each descriptor.
+        """
+        gray_image = image_utils.rgb_to_gray_cv(image)
```

Comment:
could we add a reference to the line number in COLMAP where they do this RGB->gray conversion?